lib/ansible/modules/monitoring/zabbix/zabbix_host.py

```diff
@@ -533,7 +533,7 @@ def check_all_properties(self, host_id, host_groups, status, interfaces, templat
 
         # Check whether the visible_name has changed; Zabbix defaults to the technical hostname if not set.
         if visible_name:
-            if host['name'] != visible_name and host['name'] != host_name:
+            if host['name'] != visible_name and host['host'] != visible_name:
```

Comment:
Please let me know what do you think about this:

```suggestion
            if host['name'] != visible_name:
```

Reply:
for me it also works, but the second condition avoids  changing zabbix visible name into the same as the technical_name